common/modules/gcp-project/main.tf

```diff
@@ -0,0 +1,119 @@
+# this Terraform module creates the same resources as project_init script and
+# set the DNS zones structure.
+
+terraform {
+  backend "gcs" {}
+}
+
+variable "organization_name" {
+  default = "gpii"
+}
+
+variable "organization_domain" {
+  default = "gpii.net"
+}
+
+variable "project_name" {} # name of the project to create
+
+variable "billing_id" {}
+
+variable "organization_id" {}
+
+variable "serviceaccount_key" {}
+
+variable "project_id" {} # id of the project which owns the credentials used by the provider
+
+
+provider "google" {
+  credentials = "${var.serviceaccount_key}"
+  project     = "${var.project_id}"
+  region      = "us-central1"
+}
+
+
+locals  {
+  dnsname = "${replace(
+              replace(
+                google_project.project.name,
+                "/([\\w]+)-([\\w]+)-([\\w]+)-?([\\w]+)?/",
+                "$4.$3.$2.${var.organization_domain}."),
```

Comment:
I would expect `google_project.project.name` to already have the organization domain (`gpii` or `testDomain` or whatever). Hence, replacing that value with `$(var.organization_domain}` seems incorrect vs using `$1`. Am I missing something here?

Reply:
The name of the testing organization is "gpii2test" and the domain used for that organization is "test1.gpii.net". A sample of the name of the projects created is `gpii2test-gcp-dev-joe`, which doesn't match with the domain name.

In the case of the GPII projects, the project name includes the `gpii` string of the `gpii.net`, we assumed the `.net`. That's why I introduce the `organization_domain` in the code, because the name of the organization may not match the domain name, and we use the name of the organization to give a name for the projects.